### PR TITLE
Fix deploy issues when using Juju 2.

### PR DESCRIPTION
**Pull request by frankban**
Fix "make deploy" to deploy the charm from a path rather than a local repo.
Remove no longer required files and definitions.
Avoid using the system site packages when creating the Python virtualenv.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/127/
Test PASSed.

File: tests/00-setup [hunk 1]
```diff
@@ -33,14 +35,16 @@ createvenv() {
...
-	# release of python-apt that installs via pip on xenial.
-        virtualenv --system-site-packages --distribute --extra-search-dir=$DOWNLOADCACHE $VENV
+        virtualenv --distribute --extra-search-dir=$DOWNLOADCACHE $VENV
         . $ACTIVATE && \
         pip install --upgrade pip && \
         pip install --find-links releases --find-links jujugui-deps jujugui --no-cache && \
-        BZR_PLUGIN_PATH='-user' pip install -r $TEST_REQUIREMENTS --find-links releases --find-links deps --find-links $DOWNLOADCACHE --no-allow-external --no-index --no-cache
+        BZR_PLUGIN_PATH='-user' pip install -r $TEST_REQUIREMENTS --find-links releases --find-links deps --find-links $DOWNLOADCACHE --no-allow-external --no-index --no-cache && \
+        ln -s $APT_PATH $VENV/lib/python2.7/site-packages/ && \
+        ln -s $APT_PKG_PATH $VENV/lib/python2.7/site-packages/
```
Comment (bac) [inline tests/00-setup]: Nice work around. Thanks.

File: tests/test_deploy.py [hunk 2]
```diff
@@ -17,101 +17,28 @@
...
+from deploy import juju_deploy
 
 
 def contains(needle, haystack):
     """Does the sequence (haystack) contain the given subsequence (needle)?"""
-    for i in xrange(len(haystack)-len(needle)+1):
-        for j in xrange(len(needle)):
+    for i in range(len(haystack)-len(needle)+1):
+        for j in range(len(needle)):
             if haystack[i+j] != needle[j]:
```
Comment (bac) [inline tests/test_deploy.py]: Curious why you made this change.

**Comment by bac** [comment]
:+1:
QA-OK

Note that we're leaving juju-1 in dust. 👏

File: tests/test_deploy.py (hunk 2 above)
Comment (frankban) [inline tests/test_deploy.py]: For two reasons, not particularly strong:
1) xrange is no longer present in Python3, so using range is better in the unlikely case we'll switch the charm to Python3;
2) for such small ranges, a generator is probably overkill anyway, and slightly slower.

**Comment by hatched** [comment]
:+1: Thanks, nice cleanup!

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**